drivers/clock_control/clock_control_mcux_ccm.c

```diff
@@ -135,6 +135,13 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 		break;
 #endif
 
+#ifdef CONFIG_I2S_MCUX_SAI
+	case IMX_CCM_SAI1_CLK:
+	case IMX_CCM_SAI2_CLK:
+	case IMX_CCM_SAI3_CLK:
+		*rate = CLOCK_GetFreq(kCLOCK_AudioPllClk);
+		break;
```

Comment:
Why not read the prediv and divider values her and return the final clock rate. Have you also looked at the API `CLOCK_GetClockOutClkO2Freq(kCLOCK_OutputSai1Clk)` provided by the SDK.

Reply:
ok, I can add those code. This API(uint32_t CLOCK_GetClockOutClkO2Freq(void)) is to set the out2 from given source, which is used to debug the clock source.  it first selects the clock root, and then divides the output dividor.